### PR TITLE
Scenario related schema

### PR DESCRIPTION
**Pull request by ktbee**
@gnarf @evmiguel here at the schema for adding tables we need for creating and running scenarios. I think it's pretty self-explanatory, but let me know if you have any questions/suggestions. I've been testing it using `psql` in my terminal.

### CODE REVIEW

File: src/server/migrations/20190918172031-scenarios.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+exports.up = function(db) {
+    return db.runSql(
+        `
+CREATE OR REPLACE FUNCTION updated_at() RETURNS TRIGGER AS $$
```
Comment (gnarf) [inline src/server/migrations/20190918172031-scenarios.js]: if we split all the tables up - i might suggest we split this up into it's own "functions" too

File: src/server/migrations/20190918172031-scenarios.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  BEGIN
+    IF (NEW != OLD) THEN
```
Comment (evmiguel) [inline src/server/migrations/20190918172031-scenarios.js]: When would the columns be different, such that you had to add and updated_at column? Is there a scenario where an item is created without an updated_at field?

Comment (ktbee) [reply src/server/migrations/20190918172031-scenarios.js]: Currently we don't set the updated_at field when an item is created, so it's NULL until a scenario row is updated. Then if there are any changes to any of a scenario's fields, we'll record the time the change was made. This if-statement check is confirming that the new data in the column is actually different from the old data in the column. 

Let me know if that doesn't cover your question!

**Comment by ktbee** [comment]
@evmiguel good call! I've merged in #30 and I'll rebase this branch to take advantage of it.

File: src/server/migrations/20190918172031-scenarios.js (hunk 2 above)
Comment (gnarf) [reply src/server/migrations/20190918172031-scenarios.js]: It's the same `updated_at` trigger from bocoup API -- It's a plpgsql script that checks that the "NEW" content is different from the "OLD" content, and if it is, sets the NEW.updated_at = CURRENT_TIMESTAMP before it's saved to the DB -- effectively, `UPDATE table SET value='foo'` automtically also does `SET updated_at = CURRENT_TIMESTAMP` but only if the UPDATE actually changed something

**Comment by ktbee** [comment]
@gnarf @evmiguel I've incorporated Corey's SQL changes and a request for more timestamps from Garron and Justin. Mind taking another look?

File: src/server/migrations/sql/20190919172555-update-at-function.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+CREATE OR REPLACE FUNCTION updated_at() RETURNS TRIGGER AS $$
```
Comment (gnarf) [inline src/server/migrations/sql/20190919172555-update-at-function.sql]: ordering wise - this should be first -- just rename the date part of the filenames -- it uses that to "order" them
this needs to come before something that uses it.

**Pull request merged**